src/interfaces/toMarcRecords.js

```diff
@@ -0,0 +1,74 @@
+import {Json, MARCXML, AlephSequential, ISO2709} from '@natlibfi/marc-record-serializers';
+import {Utils} from '@natlibfi/melinda-commons';
+import {amqpFactory, OPERATIONS} from '@natlibfi/melinda-rest-api-commons';
+import {AMQP_URL} from '../config';
+
+const {createLogger, toAlephId} = Utils;
+
+export async function streamToMarcRecords({correlationId, headers, stream}) {
+	const {operation, contentType} = headers;
+	const logger = createLogger();
+	const amqpOperator = await amqpFactory(AMQP_URL);
+	let recordNumber = 0;
+	let promises = [];
+	const reader = chooseAndInitReader();
+
+	// Purge queue before importing records in
+	await amqpOperator.checkQueue(correlationId, 'messages', true);
+
+	await new Promise((resolve, reject) => {
+		reader.on('error', err => {
+			reject(err);
+		}).on('data', data => {
```

Comment:
Catch errors in this block and call `reject` or they won't be handled properly

Reply:
I'll catch it in the app.js to be able to set queue item state easyer